test/extended/build_docker_test.go

```diff
@@ -0,0 +1,68 @@
+package extended
+
+import (
+	"testing"
+	"time"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	buildapi "github.com/openshift/origin/pkg/build/api"
+	"github.com/openshift/origin/test/util"
+)
+
+func init() {
+	util.RequireDocker()
+	util.RequireServer()
+}
+
+func TestBuildDocker(t *testing.T) {
+	build := mockBuildDocker()
+	client := util.GetClient()
+	if _, err := client.Builds("extendeddocker").Create(build); err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+
+	retries := 0
+
+	for {
+		time.Sleep(time.Duration(1 * time.Second))
+		current, err := client.Builds("extendeddocker").Get("sample")
+		if err != nil {
+			t.Fatalf("Unexpected error: %v", err)
+		}
+		switch current.Status {
+		case buildapi.BuildStatusFailed, buildapi.BuildStatusError:
+			t.Fatalf("Unexpected failure of build: %v", current)
+		case buildapi.BuildStatusComplete:
```

Comment:
again completion!=success.  I think we need a mock sti + mock custom builder that 1) runs super (trivial assemble script) and 2) produces an output that can easily be validated (less sure how we can easily do that w/o having to startup up the resulting container, but even if we have to start the output container, we can start it via docker directly rather than via k8s, which saves some time...no scheduler/deployer/etc)